ESCPOS_NET/Utils/DeviceFinder.cs

```diff
@@ -0,0 +1,211 @@
+        {
+            key.fmtid = new Guid(l, w1, w2, b1, b2, b3, b4, b5, b6, b7, b8);
+            key.pid = pid;
+        }
+        #endregion
+        #region Device Methods
+        public static List<DeviceDetails> GetDevices()
+        {
+            //https://learn.microsoft.com/en-us/windows-hardware/drivers/install/guid-devinterface-usb-device
+            //USB devices: a5dcbf10-6530-11d2-901f-00c04fb951ed
+            //Bluetooth devices: 00f40965-e89d-4487-9890-87c3abb211f4
+            var devices = GetDevicesbyClassID("a5dcbf10-6530-11d2-901f-00c04fb951ed");
+            return devices;
+        }
+        private static List<DeviceDetails> GetDevicesbyClassID(string classguid)
+        {
+            IntPtr intPtr = IntPtr.Zero;
+            var devices = new List<DeviceDetails>();
+            try
+            {
+                Guid guid = new(classguid);
+                intPtr = SetupDiGetClassDevs(ref guid, IntPtr.Zero, IntPtr.Zero, 18);
+                if (intPtr == INVALID_HANDLE_VALUE)
+                {
+                    Win32Exception("Failed to enumerate devices.");
+                }
+
+                int num = 0;
+                while (true)
+                {
+                    SP_DEVICE_INTERFACE_DATA DeviceInterfaceData = default;
+                    DeviceInterfaceData.cbSize = Marshal.SizeOf((object)DeviceInterfaceData);
+                    if (!SetupDiEnumDeviceInterfaces(intPtr, IntPtr.Zero, ref guid, num, ref DeviceInterfaceData))
+                    {
+                        break;
+                    }
+
+                    int RequiredSize = 0;
+                    if (!SetupDiGetDeviceInterfaceDetail(intPtr, ref DeviceInterfaceData, IntPtr.Zero, 0, ref RequiredSize, IntPtr.Zero) && Marshal.GetLastWin32Error() != 122)
+                    {
+                        Win32Exception("Failed to get interface details buffer size.");
+                    }
+
+                    IntPtr intPtr2 = IntPtr.Zero;
+                    try
+                    {
+                        intPtr2 = Marshal.AllocHGlobal(RequiredSize);
+                        Marshal.WriteInt32(intPtr2, (IntPtr.Size == 4) ? (4 + Marshal.SystemDefaultCharSize) : 8);
+                        SP_DEVINFO_DATA DeviceInfoData = default;
+                        DeviceInfoData.cbSize = Marshal.SizeOf((object)DeviceInfoData);
+                        if (!SetupDiGetDeviceInterfaceDetail(intPtr, ref DeviceInterfaceData, intPtr2, RequiredSize, ref RequiredSize, ref DeviceInfoData))
+                        {
+                            Win32Exception("Failed to get device interface details.");
+                        }
+                        string path = Marshal.PtrToStringUni(new IntPtr(intPtr2.ToInt64() + 4));
+
+                        DeviceDetails deviceDetails = GetDeviceDetails(path, intPtr, DeviceInfoData);
```

Comment:
For some devises (non printers) I got `System.ComponentModel.Win32Exception (1168)` here. 

Maybe it will be a good idea to just to skip problematic ones (try-catch-continue?) and go on instead of throwing an exception and failing to enumerate any of them.